src/replicache.ts

```diff
@@ -733,7 +733,7 @@ export class Replicache<MD extends MutatorDefs = {}>
             ? this.getAuth()
             : // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
               this.getPushAuth!());
-          if (auth != null) {
+          if (auth !== null) {
```

Comment:
Please be esp. careful in reviewing this change and the one at 803, as I'm not precisely sure what condition we mean to be testing for here. Thanks!

Reply:
This needs to be `!= null` or `auth !== null && auth !== undefined` because the type def for `getAuth` allows returning both `null` and `undefined`.

Reply:
Will change to encompass both. That means that once `getAuth()` has returned a non-null/undefined value it can never change `this.auth` back to undefined, and I wasn't sure if that was the intention or not. :)

Reply:
In this case, can it just be `if (auth) { ...`?